src/uct/tcp/tcp_ep.c

```diff
@@ -554,7 +642,11 @@ static unsigned uct_tcp_ep_progress_data_tx(uct_tcp_ep_t *ep)
     ucs_trace_func("ep=%p", ep);
 
     if (uct_tcp_ep_ctx_buf_need_progress(&ep->tx)) {
-        count += uct_tcp_ep_send(ep);
+        if (!(ep->ctx_caps & UCS_BIT(UCT_TCP_EP_CTX_TYPE_ZCOPY_TX))) {
```

Comment:
is it possible there will be a mix of zcopy and non-zcopy on same ep?

Reply:
only one operation can be in progress at the same time